pkg/kubectl/cmd/delete.go

```diff
@@ -248,13 +261,20 @@ func ReapResult(r *resource.Result, f cmdutil.Factory, out io.Writer, isDefaultD
 		if err != nil {
 			return err
 		}
+		// Set WaitForDeletion to false for namespace deletion unless it was explicitly set by the user.
+		if info.Mapping.Resource == "Namespace" && isDefaultWaitForDeletion {
```

Comment:
Different apiservers can have different behavior.  The API does not guarantee that the resource is actually deleted when DELETE request to apiserver returns. For ex: kube-apiserver and federation-apiserver have different behavior.

kubectl, on the other hand, (as discussed in https://github.com/kubernetes/kubernetes/pull/40714#issuecomment-278460041) should guarantee that the resource is deleted. I agree, that ideally this should be consistent for all resources, but we cant break namespaces now. So that is an exception.

cc @bgrant0607 @smarterclayton